ote/src/clj/ote/services/service_search.clj

```diff
@@ -192,18 +195,19 @@ Negative return value is an invalid match"
   `offset` and `limit` make it possible to use paging instead of fetching all the results"
   [db ids operation-area offset limit]
   (let [match-qualities (service-search-match-qualities db ids operation-area)
-        sorted-ids (sort-by match-qualities (keys match-qualities))
-        page (page-of sorted-ids offset limit)
-        results (transport-services db page)
+        sorted-ids (sort-by (juxt :difference :modified) match-qualities)
+        paged-result (page-of sorted-ids offset limit)
+        results (transport-services db paged-result)
         results (map
                   (fn [result]
-                    (let [service-id (::t-service/id result)
-                          matching-value (get match-qualities service-id)]
-                      (assoc result ::t-service/difference matching-value)))
+                    (let [maps-by-id (zipmap (map :id match-qualities) match-qualities)
```

Comment:
Creation of `maps-by-id` should be outside the (map ... results) because single result does not affect `maps-by-id`. Thus now `maps-by-id` is re-created each iteration identically when one symbol definition should be enough.

Reply:
Moved